### ISSUE TITLE
podman search differs from docker search behavior

### ISSUE THREAD

**Issue by thomasmckay** [projectatomic/libpod#977]
If podman is meant to be a "drop-in replacement for docker" then I'd expect this to work:
```
$ sudo docker search devel.example.com/
INDEX         NAME                                 DESCRIPTION   STARS     OFFICIAL   AUTOMATED
example.com   devel.example.com/alpine                           0                    
example.com   devel.example.com/busybox                          0                    
example.com   devel.example.com/coreos/alpine-sh                 0

$ sudo podman search devel.example.com/
ERRO[0001] error searching registry "registry.access.redhat.com": error creating new docker client: open /etc/docker/certs.d/registry.access.redhat.com/redhat-ca.crt: no such file or directory 
```

I was told that I needed to use the _--registry_ option with _podman search_ but that is non-conformant with expectations.

```
$ podman version
Version:       0.4.1
Go Version:    go1.9.4
OS/Arch:       linux/amd64
```

**Comment by rhatdan** [comment]
@haircommander PTAL
podman search should go through the list of registries and search each one of them.  Not just stop at the first.

**Comment by haircommander** [comment]
@thomasmckay can you show me the contents of your /etc/containers/registries.conf file?

**Comment by thomasmckay** [comment]
```
$ sudo cat /etc/containers/registries.conf 
# This is a system-wide configuration file used to
# keep track of registries for various container backends.
# It adheres to TOML format and does not support recursive
# lists of registries.

# The default location for this configuration file is /etc/containers/registries.conf.

# The only valid categories are: 'registries.search', 'registries.insecure', 
# and 'registries.block'.

[registries.search]
registries = ['docker.io','registry.centos.org','registry.access.redhat.com']

# If you need to access insecure registries, add the registry's fully-qualified name.
# An insecure registry is one that does not have a valid SSL certificate or only does HTTP.
[registries.insecure]
registries = []


# If you need to block pull access from a registry, uncomment the section below
# and add the registries fully-qualified name.
#
# Docker only
[registries.block]
registries = []
```

**Comment by rhatdan** [comment]
@haircommander Need to vendor in latest containers/image into libpod.
https://github.com/containers/image/pull/470

**Comment by umohnani8** [comment]
It does search through all the registries in the registries.conf file by default. The `--registry` flag is optional if you just want to search a specific registry. If the image is not found in a registry there will be 0 results for that registry.

**Comment by rhatdan** [comment]
I would like to remove th --registry flag or at least hide it, since to me it makes little sense.

**Comment by umohnani8** [comment]
But what if you don't want ten million results because you have like 6 default registries and just want to search for an image in one registry? I can make the man page better.
I like this option because it makes search faster if I just want to search on one registry instead of waiting for results from all the default registries.

**Comment by rhatdan** [comment]
I just think it is hard to discover, I am fine with it.

**Comment by thomasmckay** [comment]
What is the difference between these two patterns?

_podman search --registry devel.example.com something_
_podman search devel.example.com/something_

The second one just implies a single registry and is the established pattern for existing _docker_ users. I'm find w/ searching all _registry.conf_ registries if no registry specified, but if I prefix the search with a registry it seems sensible just to search that single one.

**Comment by umohnani8** [comment]
I actually had no clue that docker search did that. The help says "Search the Docker Hub for images" so i thought its search was limited to docker.io.
Will work on matching that.

**Comment by TomSweeneyRedHat** [comment]
@thomasmckay +1 on the thought and thanks for the questions.  Ooop, speedy @umohnani8 beat me to the punchline though!

**Comment by umohnani8** [comment]
Fixed in #1011
